src/halmos/sevm.py

```diff
@@ -1979,78 +2159,75 @@ def create(
                 st=State(),
                 jumpis={},
-                output=None,
                 symbolic=False,
                 prank=Prank(),
                 #
                 solver=ex.solver,
                 path=ex.path,
                 alias=ex.alias,
                 #
-                log=ex.log,
                 cnts=ex.cnts,
                 sha3s=ex.sha3s,
                 storages=ex.storages,
                 balances=ex.balances,
                 calls=ex.calls,
-                failed=ex.failed,
-                error=ex.error,
             )
         )
 
         logs.extend(new_logs)
 
         # process result
         for idx, new_ex in enumerate(new_exs):
-            # sanity checks
-            if new_ex.failed:
-                raise ValueError(new_ex)
-
-            opcode = new_ex.current_opcode()
-            if opcode in [EVM.STOP, EVM.RETURN]:
-                # new contract code
-                new_ex.code[new_addr] = Contract(new_ex.output)
-
-                # restore tx msg
-                new_ex.calldata = ex.calldata
-                new_ex.callvalue = ex.callvalue
-                new_ex.caller = ex.caller
-                new_ex.this = ex.this
+            subcall = new_ex.context
 
-                # restore vm state
-                new_ex.pgm = ex.pgm
-                new_ex.pc = ex.pc
-                new_ex.st = deepcopy(ex.st)
-                new_ex.jumpis = deepcopy(ex.jumpis)
-                new_ex.output = None  # output is reset, not restored
-                new_ex.symbolic = ex.symbolic
-                new_ex.prank = deepcopy(ex.prank)
+            # continue execution in the context of the parent
+            # pessimistic copy because the subcall results may diverge
+            new_ex.context = deepcopy(ex.context)
+            new_ex.context.trace.append(subcall)
+
+            new_ex.this = ex.this
+
+            # restore vm state
+            new_ex.pgm = ex.pgm
+            new_ex.pc = ex.pc
+            new_ex.st = deepcopy(ex.st)
+            new_ex.jumpis = deepcopy(ex.jumpis)
+            new_ex.symbolic = ex.symbolic
+            new_ex.prank = deepcopy(ex.prank)
+
+            if subcall.output.error is None:
+                # new contract code, will revert if data is None
+                new_ex.code[new_addr] = Contract(subcall.output.data)
 
                 # push new address to stack
                 new_ex.st.push(uint256(new_addr))
 
-                # add to worklist
-                new_ex.next_pc()
-                stack.append((new_ex, step_id))
+            elif isinstance(subcall.output.error, HalmosException):
+                # abort the current path
+                out.append(new_ex)
+                continue
+
             else:
                 # creation failed
-                out.append(new_ex)
+                new_ex.st.push(con(0))
+
+            # add to worklist
+            new_ex.next_pc()
+            stack.append((new_ex, step_id))
```

Comment:
todo: now that the create failure semantics is supported, this for-loop can be made consistent with the one for `call_known()`.  further, their common logic can be factored out to avoid duplication.

Reply:
agreed, do you mind if we do a cleanups as a subsequent PR though in the interest of getting this merged faster?